sympy/sets/fancysets.py

```diff
@@ -308,7 +312,10 @@ def __new__(cls, flambda, *sets):
             return sets[0]
 
         if not set(flambda.variables) & flambda.expr.free_symbols:
-            return FiniteSet(flambda.expr)
+            if all(s.is_empty for s in sets):
+                return S.EmptySet
+            elif any(s.is_empty == False for s in sets):
+                return FiniteSet(flambda.expr)
```

Comment:
ImageSet is very poorly defined - I plan to fix it. I'm not sure about the logic above though.

The situation here is that we have an ImageSet with a constant map. The result is empty set if the baseset is empty and FiniteSet otherwise. However I think we have to interpret the base set as the productset of the plural sets in which case if any is empty then the whole thing is empty:
```julia
In [11]: ProductSet(S.EmptySet, Interval(0, 1))                                                                                   
Out[11]: ∅
```
I think the best thing would be to use `self.baseset.is_empty` i.e. defer the logic to ProductSet. You can't really use `self.baseset` since this is a classmethod but it should be equivalent.

Reply:
Ouch, I was fixated on single-variable ImageSets and must have thought "Oh, it maps multiple sets to their image, ok then at least one must be non-empty". I completely forgot about multi-variable ImageSets and Lambdas...

I'll change it to:
```
        if not set(flambda.variables) & flambda.expr.free_symbols:
            if any(s.is_empty for s in sets):
                return S.EmptySet
            elif all(s.is_empty == False for s in sets):
                return FiniteSet(flambda.expr)
```